File: files/CH-152.py
import pandas as pd

path = "CH-152 Extract from Text.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="E:H", skiprows=1, nrows=6)

def split_on_top_level(s):
    chars = list(s)
    level = 0
    current = ""
    result = []
    for char in chars:
        if char == '{':
            level += 1
        if char == '}':
            level -= 1
        if char == ',' and level == 0:
            result.append(current)
            current = ""
        else:
            current += char
    if current:
        result.append(current)
    return result

input['components'] = input['Value'].apply(split_on_top_level)
components = pd.DataFrame(input['components'].tolist())
components.columns = [f"List.{i+1}" for i in range(components.shape[1])]
result = pd.concat([input.drop(columns=['Value', 'components']), components], axis=1)
test.columns = result.columns

result = result.fillna('')
test = test.fillna('')

print(result.equals(test))  # True

File: files/100-199/152/CH-152.R
library(tidyverse)
library(readxl)

path = "files/CH-152 Extract from Text.xlsx"
input = read_excel(path, range = "B2:C7")
test  = read_excel(path, range = "E2:H7")

split_on_top_level <- function(s) {
  chars <- strsplit(s, "")[[1]]
  level <- 0
  current <- ""
  result <- c()
  for (char in chars) {
    if (char == '{') level <- level + 1
    if (char == '}') level <- level - 1
    if (char == ',' && level == 0) {
      result <- c(result, current)
      current <- ""
    } else {
      current <- paste0(current, char)
    }
  }
  if (nchar(current) > 0) result <- c(result, current)
  return(result)
}

result = input %>%
  mutate(components = map(Value, split_on_top_level)) %>%
  select(-Value) %>%
  unnest_wider(components, names_sep = "") %>%
  rename_with(~paste0("List.", seq_along(.)), starts_with("components"))

all.equal(result, test)
#> [1] TRUE